plugins/lua/buildingplan/itemselection.lua

```diff
@@ -256,33 +263,38 @@ function ItemSelection:get_choices(sort_fn)
         local item = df.item.find(item_id)
         if not item then goto continue end
         local desc = get_item_description(item_id, item)
-        if buckets[desc] then
-            local bucket = buckets[desc]
+        local value = dfhack.items.getValue(item)
+        local key = desc .. "_" .. tostring(value)
+        if buckets[key] then
+            local bucket = buckets[key]
             table.insert(bucket.data.item_ids, item_id)
             bucket.data.quantity = bucket.data.quantity + 1
         else
             local entry = {
                 search_key=make_search_key(desc),
                 icon=self:callback('get_entry_icon', item_id),
                 data={
+                    desc=desc,
                     item_ids={item_id},
                     item_type=item:getType(),
                     item_subtype=item:getSubtype(),
                     quantity=1,
                     quality=item:getQuality(),
+                    value=value,
                     selected=0,
                 },
             }
-            buckets[desc] = entry
+            buckets[key] = entry
         end
         ::continue::
     end
     local choices = {}
-    for desc,choice in pairs(buckets) do
+    for key,choice in pairs(buckets) do
         local data = choice.data
         choice.text = {
-            {width=10, text=function() return ('%d/%d'):format(data.selected, data.quantity) end},
-            {gap=2, text=desc},
+            {width=8, text=function() return ('%d/%d'):format(data.selected, data.quantity) end},
+            {width=8, gap=2, text=function() return ('%d\x0F'):format(data.value) end},
```

Comment:
displaying the value directly is a spoiler. Please use the output of `reqscript('internal/caravan/common').obfuscate_value(value)`